### DISCUSSION TITLE
Redirects after verify account from email

### DISCUSSION THREAD

**Discussion by Halvanhelv** [janko/rodauth-rails#249] [Q&A] [answered]
Hey, Janko.

I am implementing a functionality that after registration/login users would go to a certain page by redirect, I implemented it by passing an additional parameter in the link and on the backend I check, if there is a certain link I write the following into the session 

`session[login_redirect_session_key] = param('redirect_to_path')`

then Rodauth after authorization redirects me to where I want and it seems to work in 3 cases out of 4.

1) when user sign in via mail
2) when user sign in via google
3) when a user sign up via google

but it doesn't work in case when user sign up via mail because it sends an email 

I tried to add an additional parameter in url (where to redirect to) manually in the link from email for verification to test the theory, in `before_verify_account` rodauth hook I added next code:
**`session[login_redirect_session_key] = param('redirect_to_path')`**

but the problem is that when the code reaches `verify_account_redirect` then apparently this key is overwritten in the session and I can not make a redirect where I need, please tell me what can be done in this situation? Thanks

**Comment by janko** [comment, marked as answer]
In the 3 cases that you mentioned, all of them call the `#login` method, which handles `session[login_redirect_session_key]`. When the account is verified, the autologin that's performed doesn't call the `#login` method, but updates the session directly, clearing `session[login_redirect_session_key]` before `verify_account_redirect` is reached.

You could try saving it in an instance variable instead:

```rb
before_verify_account_route do
  if request.get?
    session[:verify_account_redirect] = param('redirect_to_path')
  elsif request.post?
    @verify_account_redirect = session[:verify_account_redirect]
  end
end

verify_account_redirect { @verify_account_redirect || super() }
```

**Reply by Halvanhelv** [reply]
Thanks for the help, it worked
